controls/mousestates/cMousePlaceState.cpp

```diff
@@ -82,8 +82,8 @@ bool cMousePlaceState::mayPlaceIt(cBuildingListItem *itemToPlace, int mouseCell)
     int iEndY = iCellY + SCANWIDTH + cellHeight;
 
     // Fix up the boundaries
-    FIX_POS(iStartX, iStartY);
-    FIX_POS(iEndX, iEndY);
+    map.fixCoordinatesToBeWithinMap(iStartX, iStartY);
```

Comment:
tbh, I really hate this function. As it changes the values using out vars. I'd rather have some function that returns corrected values (even though it is a bit more verbose). Anyway, for now this is 👌

Reply:
with "this function" i meant "FIX_POS()" ;-)

Reply:
Output parameters are indeed frowned upon nowadays. But with C++17 and the "structured bindings" feature, we can return multiple values. I'll see what I can do in this PR.